website/client/src/libs/analytics.js

```diff
@@ -63,15 +66,24 @@ function _gatherUserStats (properties) {
   if (user.purchased.plan.planId) properties.subscription = user.purchased.plan.planId;
 }
 
-export function setUser () {
-  const user = _getConsentedUser();
-  if (!user) return;
-  amplitude.getInstance().setUserId(user._id);
+export function setup (userId) {
+  if (analyticsLoading) return;
```

Comment:
idea: Instead of having the `if (analyticsReady) {}` check in each calling method, how about changing this line to

`if (analyticsReady || analyticsLoading)`

Simplifies the code somewhat. To make understanding easier, method coulda also be renamed to `safeSetup` or something to indicate it prevents duplicate setup runs.